packages/go/test/parse-go-mod-version.test.ts

```diff
@@ -0,0 +1,38 @@
+import { parseGoModVersion } from '../src/go-helpers';
+
+describe('parseGoModVersion', function () {
+  it('returns undefined with empty string', async () => {
+    const version = parseGoModVersion('');
+    expect(version).toBeUndefined();
+  });
+  it('returns exactly same version if patch exists', async () => {
+    const version = parseGoModVersion('go 1.16.1');
```

Comment:
This is kind of an invalid test case.
Go only supports a patch version starting with go 1.21.
Before that, having a patch version in the go mod file is invalid 😇 

I know its just an test, but I think this should reflect the real world anyways.

Reply:
```suggestion
    const version = parseGoModVersion('go 1.21.1');
```